Zend/zend_interfaces.stub.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+interface Traversable {}
+
+interface IteratorAggregate extends Traversable
+{
+    /** @return Traversable */
+    function getIterator();
+}
+
+interface Iterator extends Traversable
+{
+    function current();
+
+    /** @return void */
+    function next();
+
+    function key();
+
+    /** @return bool */
+    function valid();
+
+    /** @return void */
+    function rewind();
+}
+
+interface ArrayAccess
+{
+    function offsetExists($offset);
+
+    /* actually this should be return by ref but atm cannot be */
```

Comment:
I'm surprised we're not getting any ref variance issues due to this.

Reply:
Ah, return refs are covariant, only arg refs are invariant, so this makes sense.